src/drivers/rt1020.c

```diff
@@ -145,14 +167,18 @@ static size_t mg_tcpip_driver_rt1020_tx(const void *buf, size_t len,
 }
 
 static bool mg_tcpip_driver_rt1020_up(struct mg_tcpip_if *ifp) {
-  uint32_t bsr = eth_read_phy(PHY_ADDR, PHY_BSR);
+  struct mg_tcpip_driver_rt1020_data *d =
+      (struct mg_tcpip_driver_rt1020_data *) ifp->driver_data;
+  uint32_t bsr = rt10xx_phy_read(d->phy_addr, PHY_BSR);
   bool up = bsr & BIT(2) ? 1 : 0;
   if ((ifp->state == MG_TCPIP_STATE_DOWN) && up) {  // link state just went up
-    uint32_t pc1r = eth_read_phy(PHY_ADDR, PHY_PC1R);
-    uint32_t tcr = ENET->TCR |= BIT(2);        // Full-duplex
-    uint32_t rcr = ENET->RCR &= ~BIT(9);       // 100M
-    if ((pc1r & 3) == 1) rcr |= BIT(9);        // 10M
-    if ((pc1r & BIT(2)) == 0) tcr &= ~BIT(2);  // Half-duplex
+    uint32_t tcr = ENET->TCR |= BIT(2);             // Full-duplex
+    uint32_t rcr = ENET->RCR &= ~BIT(9);            // 100M
+    if (rt10xx_phy_read(d->phy_addr, PHY_ID2)) {    // KSZ8081RNB ?
+      uint32_t pc1r = rt10xx_phy_read(d->phy_addr, 30);  // Read PC2R
+      if ((pc1r & 3) == 1) rcr |= BIT(9);                // 10M
+      if ((pc1r & BIT(2)) == 0) tcr &= ~BIT(2);          // Half-duplex
+    }
```

Comment:
Here we must check PHY ID again. In fact, this code, as is, will not run at 10baseT nor half-duplex on the Teensy.
For the DP83825, the register to read in order to get this info is PHYSTS (Offset = 0x10)

Actions here, are "up" and will be done by all drivers (with non built-in PHY), though each one will have its own way to access the PHY.

Reply:
Ack. I'll leave this for you!